lib/utils.js

```diff
@@ -423,6 +428,24 @@ function mergeSettings(target, src) {
 }
 
 /**
+ * TODO jsdoc
+ */
+function rankArrayElements(array, rank) {
+  if (!Array.isArray(array) || !Number.isFinite(rank))
+    return array;
+
+  array.forEach(function(el) {
+    // only apply ranking on objects {} in array
+    if (!el || typeof el != 'object' || Array.isArray(el))
+      return;
+
+    // initialize element rank if not set already
+    el.__rank = el.__rank || rank;
```

Comment:
actually I did applied @bajtos 's proposal, it's safer to prevent people interacting with this new internal property as it will be used in ACL scoring at runtime